### PR TITLE
Collect2 app passes strict null checks

### PR DESCRIPTION
**Pull request by eessex**
An experiment related to https://artsyproduct.atlassian.net/browse/PLATFORM-1780

Turned on strict null checking for typescript, and converted the `Apps/Collect2` directory to pass all checks. I've disabled strict null checking for the moment so there is a chance these changes could be accidentally removed. 

Next steps: 
- implement a strategy to turn on type-checking selectively by path.
- Develop list of apps that have greatest need for checking so we can prioritize these changes.

### CODE REVIEW

File: src/Apps/Collect2/Components/SeoProductsForArtworks.tsx [hunk 1]
```diff
@@ -29,72 +29,108 @@ export class SeoProducts extends React.Component<SeoProductsProps> {
...
+              partner &&
+              partner.profile &&
+              partner.profile.icon &&
+              partner.profile.icon.url
+            return is_acquireable && iconUrl
+          } else {
+            return false
+          }
+        })
+      })
```
Comment (damassi) [inline src/Apps/Collect2/Components/SeoProductsForArtworks.tsx]: Why not use our `get` utility throughout? The verbosity can be reduced

Comment (alloy) [reply src/Apps/Collect2/Components/SeoProductsForArtworks.tsx]: Isn’t this all already wrapped inside a `get` block, in which case you could perhaps indeed reduce the verbosity by using the `!` operator to remove `null` from these intermediate values. i.e. `partner!.profile!.icon!.url!`

Comment (alloy) [reply src/Apps/Collect2/Components/SeoProductsForArtworks.tsx]: FWIW I think it’s fine to use the verbose version in places where you’re not already using `get` to wrap a larger block, no need to imo to replace it everywhere.

**Comment by artsyit** [comment]
:rocket: PR was released in v18.1.10 :rocket:

**Pull request merged**